src/mem_table.rs

```diff
@@ -248,223 +304,142 @@ impl KVTable {
     }
 }
 
-fn sizeof_attributes(attrs: &RowAttributes) -> usize {
-    attrs.ts.map(|_| 8).unwrap_or(0)
-}
-
 #[cfg(test)]
 mod tests {
```

Comment:
Should we also have a test stressing multiple versions of the same key, including different types (tombstones, merges, ...)?

Reply:
i added the test cases over same key + different types in the merge iterator side.

memtable's `iter()` simply dump the duplicated keys inside it, the only place handles deduplication is the `MergeIterator`.